fbw-common/src/wasm/systems/systems/src/electrical/ram_air_turbine.rs

```diff
@@ -0,0 +1,188 @@
+use crate::{
+    shared::{
+        interpolation, AngularSpeedSensor, EmergencyElectricalRatPushButton,
+        EmergencyElectricalState, EmergencyGeneratorControlUnit, EmergencyGeneratorPower,
+        LgciuWeightOnWheels, RamAirTurbineController,
+    },
+    simulation::{
+        InitContext, SimulationElement, SimulationElementVisitor, SimulatorWriter, UpdateContext,
+        VariableIdentifier, Write,
+    },
+    wind_turbine::WindTurbine,
+};
+
+use uom::si::{
+    angle::radian,
+    angular_velocity::{radian_per_second, revolution_per_minute},
+    f64::*,
+    power::watt,
+    torque::newton_meter,
+};
+
+pub struct RamAirTurbine {
+    stow_position_id: VariableIdentifier,
+
+    deployment_commanded: bool,
+
+    wind_turbine: WindTurbine,
+    position: f64,
+}
+impl RamAirTurbine {
+    // Speed to go from 0 to 1 stow position per sec. 1 means full deploying in 1s
+    const STOWING_SPEED: f64 = 1.;
+
+    // Factor used to compute resistant torque. 1.1 means resistant torque is 10% higher than elec power we generate
+    //  thus a 90% efficiency
+    const GENERATOR_EFFICIENCY_PENALTY: f64 = 1.1;
+
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            stow_position_id: context.get_identifier("RAT_STOW_POSITION".to_owned()),
+
+            deployment_commanded: false,
+
+            wind_turbine: WindTurbine::new(context),
+            position: 0.,
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        controller: &impl RamAirTurbineController,
+        generator_power: &impl EmergencyGeneratorPower,
+    ) {
+        // Once commanded, stays commanded forever
+        self.deployment_commanded = controller.should_deploy() || self.deployment_commanded;
+
+        self.update_position(context);
+
+        self.update_physics(context, generator_power);
+    }
+
+    pub fn update_physics(
+        &mut self,
+        context: &UpdateContext,
+        generator_power: &impl EmergencyGeneratorPower,
+    ) {
+        let resistant_torque = self.resistant_torque(generator_power);
+        self.wind_turbine.update(
+            &context.delta(),
+            context.indicated_airspeed(),
+            self.position,
+            resistant_torque,
+        );
+    }
+
+    fn update_position(&mut self, context: &UpdateContext) {
+        if self.deployment_commanded {
+            self.position += context.delta_as_secs_f64() * Self::STOWING_SPEED;
+
+            // Finally limiting pos in [0:1] range
+            if self.position < 0. {
+                self.position = 0.;
+            } else if self.position > 1. {
+                self.position = 1.;
+            }
+        }
+    }
+
+    fn resistant_torque(&mut self, generator_power: &impl EmergencyGeneratorPower) -> Torque {
+        let mut pump_torque = 0.;
+        if self.wind_turbine.is_low_speed() {
+            pump_torque += (self.wind_turbine.position().get::<radian>() * 4.).cos() * 0.35 * 2.;
```

Comment:
Magic like you said :D
Changing those not the purpose of this PR, but idea is at low speed you have a torque added or substracted depending on the angular position and the number of hydraulic pump plunger per rotation. So if you have  like 4 plungers it will fight to reach the max compression 4 times per turn or be pushed by it if it can pass the max compression. Just like a piston engine would stall.

Now this was just an early test done like 2 years ago, we will now be able to tune this visually on the 380 model